src/BugsnagUnity/Client.cs

```diff
@@ -335,15 +315,16 @@ void Notify(Exception[] exceptions, HandledState handledState, Middleware callba
                 @event.AddAndroidProjectPackagesToEvent(Configuration.ProjectPackages);
             }
 
-            var report = new Report(Configuration, @event);
-
-            lock (MiddlewareLock)
+            lock (CallbackLock)
             {
-                foreach (var middleware in Middleware)
+                foreach (var onErrorCallback in Configuration.GetOnErrorCallbacks())
```

Comment:
This might already be scheduled - but one key concept behind the notifier spec is that after the `Client` has been initialized, it should create a copy of `Configuration` so that any subsequent changes made to the object are ignored. This minimises the amount of state change that Bugsnag has to worry about (e.g. we'd be assured that there is no possibility of concurrent modification to this collection). If it's not already ticketed it'd be worth planning something

Reply:
@tomlongridge plz could i get your opinion on this? Thanks

Reply:
Yes, as @fractalwrench says, that's the approach we should take. Would be worth checking the Android/Cocoa codebases to see how that's achieved there (i.e. where the cloning occurs) so do a similar thing here. I'd suggest a separate ticket/PR for this though.

Reply:
Ticket created 👍